framework/lib/components/text-field/formatted-number-input.tsx

```diff
@@ -1,32 +1,118 @@
-import React from 'react'
+import React, { ComponentType } from 'react'
+import { identity } from 'ramda'
+import {
+  InputAttributes,
+  NumberFormatValues,
+  NumericFormat,
+  SourceInfo,
+} from 'react-number-format'
 import { Input } from '../input'
 import { FormattedNumberInputProps } from './types'
 
+const presetMap = {
```

Comment:
This variable should be typed, as it's easy to make a typo here. Checkout _REFERENCE #1_

Reply:
Addictionally, since it's top-level function, in case you remove the use of it in the component or just replace `presetMap[preset]` with `prop(preset, presetMap)` everything is gonna go to shit. That's why explicit type assigning is so important: 1. you get an error on the actual place where the error occurs (not in some nested deep shit) 2. you know 100% that this block of code is valid and you don't have to rely on places where you use the variable.

Every variable, class, function should be a black box. You don't care how it is implemented, all you should care about is how you integrate it with your own logic. 
![image](https://github.com/mediatool/northlight/assets/40524044/bb9b3047-b070-414e-9c83-24015c7b99c7)